app/javascript/widget/App.vue

```diff
@@ -109,10 +119,28 @@ export default {
         this.unsetUnreadView();
         this.setUserLastSeen();
       });
+
+      bus.$on('on-campaign-view-clicked', campaignId => {
+        const { websiteToken } = window.chatwootWebChannel;
+        this.showCampaignView = false;
+        this.showUnreadView = false;
+        this.unsetUnreadView();
+        this.setUserLastSeen();
+        this.executeCampaign({ campaignId, websiteToken });
+      });
     },
     setPopoutDisplay(showPopoutButton) {
       this.showPopoutButton = showPopoutButton;
     },
+    setCampaignView() {
+      const { messageCount, activeCampaign } = this;
+      if (this.isIFrame && !isEmptyObject(activeCampaign) && !messageCount) {
```

Comment:
Would be more readable if we have constant with name suggesting what it does inside `if`.